common/docker/client.go

```diff
@@ -0,0 +1,81 @@
+package docker
+
+import (
+	"github.com/fsouza/go-dockerclient"
+	. "github.com/weaveworks/weave/common"
+	"regexp"
+)
+
+// An observer for container events
+type ContainerObserver interface {
+	ContainerDied(ident string) error
+}
+
+type Client struct {
+	*docker.Client
+}
+
+// IsContainerId returns True if the string provided is a valid container id
+func IsContainerId(idStr string) bool {
+	matched, err := regexp.MatchString("^[a-f0-9]+$", idStr)
```

Comment:
The golang way is to define a global `var` with `regexp.MustCompile`, and then call `MatchString` on that, which returns just a bool.

See, e.g. https://github.com/weaveworks/weave/blob/master/proxy/proxy.go#L21 and https://github.com/weaveworks/weave/blob/master/proxy/proxy.go#L89